configure

```diff
@@ -1431,6 +1420,7 @@ Optional Features:
   --enable-sigfpe         Enable FloatingPointExceptions
   --disable-backtrace     Disable function backtrace
   --enable-shared         Enable building bout++ into an shared object
+  --enable-openmp         Enable building with OpenMP support
```

Comment:
There's also an entry for disable-openmp a few lines below. I think we should probably just have one of these.

Reply:
as the default is to disable it - the one below should be removed.
However, I am not sure why it is there.

Reply:
My impression is that this is a "builtin" with autoconf (e.g. see [here](https://www.spinics.net/lists/ac/msg06330.html) and [here](https://www.gnu.org/software/autoconf/manual/autoconf-2.67/html_node/Generic-Compiler-Characteristics.html)) but I can't find any good documentation making this clear.

Reply:
I guess that cannot be disabled (from /usr/share/autoconf/autoconf/c.m4):
```
AC_DEFUN([AC_OPENMP],
[
  OPENMP_[]_AC_LANG_PREFIX[]FLAGS=
  AC_ARG_ENABLE([openmp],
    [AS_HELP_STRING([--disable-openmp], [do not use OpenMP])])
  if test "$enable_openmp" != no; then
    AC_CACHE_CHECK([for $[]_AC_CC[] option to support OpenMP],
      [ac_cv_prog_[]_AC_LANG_ABBREV[]_openmp],
```

Should this be reported to `autoconf`: https://www.gnu.org/software/autoconf/autoconf.html#bug ?

Reply:
That's not a particularly great choice from the autoconf maintainers there.

Easiest thing to do is to copy the `AC_DEFUN` from that file into `m4/bout.m4`, rename it `BOUT_OPENMP`, change the `AC_ARG_ENABLE` to be what we actually want, then use our new macro instead of `AC_OPENMP`.